examples/scan.py

```diff
@@ -0,0 +1,193 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import datetime
+
+import numpy as np
+from benchmark import run_benchmark
+
+import cunumeric as num
+
+
+def initialize(shape, dt, axis):
+    if dt == "int":
+        A = np.random.randint(1000, size=shape).astype(np.int32)
+        if axis is None:
+            B = np.zeros(shape=A.size, dtype=np.int32)
+        else:
+            B = np.zeros(shape=shape, dtype=np.int32)
+    elif dt == "float":
+        A = np.random.random(shape).astype(np.float32)
+        # insert NAN at second element
+        if len(shape) == 1:
+            A[1] = np.nan
+        elif len(shape) == 2:
+            A[1, 1] = np.nan
+        elif len(shape) == 3:
+            A[1, 1, 1] = np.nan
+        elif len(shape) == 4:
+            A[1, 1, 1, 1] = np.nan
+
+        if axis is None:
+            B = np.zeros(shape=A.size, dtype=np.float32)
+        else:
+            B = np.zeros(shape=shape, dtype=np.float32)
+    else:
+        A = (
+            np.random.random(shape).astype(np.float32)
+            + np.random.random(shape).astype(np.float32) * 1j
+        )
+        if len(shape) == 1:
+            A[1] = np.nan
+        elif len(shape) == 2:
+            A[1, 1] = np.nan
+        elif len(shape) == 3:
+            A[1, 1, 1] = np.nan
+        elif len(shape) == 4:
+            A[1, 1, 1, 1] = np.nan
+
+        if axis is None:
+            B = np.zeros(shape=A.size, dtype=np.complex64)
+        else:
+            B = np.zeros(shape=shape, dtype=np.complex64)
+
+    return A, B
+
+
+def check_scan(OP, A, B, ax):
+    C = np.zeros(shape=B.shape, dtype=B.dtype)
+    if OP == "cumsum":
+        np.cumsum(A, out=C, axis=ax)
+    elif OP == "cumprod":
+        np.cumprod(A, out=C, axis=ax)
+    elif OP == "nancumsum":
+        np.nancumsum(A, out=C, axis=ax)
+    elif OP == "nancumprod":
+        np.nancumprod(A, out=C, axis=ax)
+
+    print("Checking result...")
+    if np.allclose(B, C, equal_nan=True):
+        print("PASS!")
+    else:
+        print("FAIL!")
+        print("INPUT    : " + str(A))
+        print("CUNUMERIC: " + str(B))
+        print("NUMPY    : " + str(C))
+        assert False
+
+
+def run_scan(OP, shape, dt, ax, check):
+    print("Problem Size:    shape=" + str(shape))
+
+    # axis handling
+    if ax is not None:
+        assert ax < len(shape) and ax >= 0
+
+    print("Problem Type:    OP=" + OP)
+    print("Axis:            axis=" + str(ax))
+    print("Data type:       dtype=" + dt + "32")
+    A, B = initialize(shape=shape, dt=dt, axis=ax)
```

Comment:
8ca953acd6c147188c2e8d9868f9057c77ae976d
I made the change, however, `cunumeric.random.randint()` is causing errors (same call works fine through numpy).

Reply:
I've added an issue for randint() (https://github.com/nv-legate/cunumeric/issues/444)
However for this PR to avoid dependence on the issue I've made a change to the example code to avoid CI errors
19d2b93f8aa8cd8f6076809a4c001fa37fa8c2d2